### PR TITLE
Auraescript on Deno

### PR DESCRIPTION
**Pull request by future-highway**
Current state:
- op_decl are generated with code similar to what we had with rhai (see runtime/mod.rs)
- build.rs will generate TypeScript files based on our proto files  (e.g., lib/runtime.ts) using ts-proto library from npm
- macros generate op_decls and augment the generated typescript files with clients that call the appropriate deno ops.
- TS script can import the generated typescript files (see auraescript_deno/examples/script.ts)
- confirmed can execute TS scripts (with script.ts a call is successfully sent to auraed which errors with not implemented for `CellService.allocate`)
- did not confirm JS script works

I named the crates "auraescript_deno" and "auraescript_deno_macros" just so that they don't interfere what we have or other PR's. However auraescript_deno's bin is still named auraescript

### CODE REVIEW

**Comment by dmah42** [comment]
how does this overlap with https://github.com/aurae-runtime/aurae/tree/deno ?

**Comment by dmah42** [comment]
> how does this overlap with https://github.com/aurae-runtime/aurae/tree/deno ?

discussed in the discord: this is the way.

File: auraescript/Cargo.toml [hunk 1]
```diff
@@ -1,69 +1,25 @@
...
-#   See the License for the specific language governing permissions and        #
-#   limitations under the License.                                             #                                                                             #
-#                                                                              #
-# ---------------------------------------------------------------------------- #
-
 [package]
 name = "auraescript"
 version = "0.1.0"
 edition = "2021"
-authors = ["The Aurae Authors", "Kris Nóva <kris@nivenly.com>"]
```
Comment (dmah42) [inline auraescript/Cargo.toml:35]: might be good to keep this and the above license notice

File: auraescript/LICENSE [hunk 2]
```diff
@@ -1,201 +0,0 @@
-                                 Apache License
```
Comment (dmah42) [inline auraescript/LICENSE:1]: why did this get deleted?

File: auraescript/Makefile [hunk 3]
```diff
@@ -41,7 +41,7 @@ compile: ## Compile for the local architecture ⚙
 install: ## Build and install (debug) 🎉
 	@$(cargo) clippy
 	@$(cargo) install --debug --path .
-	sudo -E cp -rv examples/bin/* /usr/bin
+	#sudo -E cp -rv examples/bin/* /usr/bin
```
Comment (dmah42) [inline auraescript/Makefile:44]: i guess either put this back or remove the line?

File: auraescript/README.md [hunk 4]
```diff
@@ -1,62 +0,0 @@
-# AuraeScript
```
Comment (dmah42) [inline auraescript/README.md:1]: is this not relevant any more?

File: auraescript/src/bin/main.rs [hunk 5]
```diff
@@ -1,169 +1,32 @@
...
- *   Unless required by applicable law or agreed to in writing, software      *
- *   distributed under the License is distributed on an "AS IS" BASIS,        *
- *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
- *   See the License for the specific language governing permissions and      *
- *   limitations under the License.                                           *
- *                                                                            *
-\* -------------------------------------------------------------------------- */
-
-// TODO @kris-nova as we move to Deno we probably want to revist the main function
-// Lint groups: https://doc.rust-lang.org/rustc/lints/groups.html
```
Comment (dmah42) [inline auraescript/src/bin/main.rs:32]: do you mind putting these back? they shouldn't be hard to resolve i hope...

File: auraescript/LICENSE (hunk 2 above)
Comment (future-highway) [reply auraescript/LICENSE]: By accident, but I'm not sure if we still need it. Is this an artifact of when we had multiple repos. Put back for now.

File: auraescript/README.md (hunk 4 above)
Comment (future-highway) [reply auraescript/README.md:1]: I put it back and made some updates to make it more inline with what we have.

File: auraescript/Cargo.toml [hunk 6]
```diff
@@ -1,69 +1,26 @@
...
+thors = ["The Aurae Authors", "Kris Nóva <kris@nivenly.com>"]
```
Comment (dmah42) [inline auraescript/Cargo.toml:5]: "we're still worthy!" but i think this should be "authors"

File: auraescript/src/bin/main.rs (hunk 5 above)
Comment (future-highway) [reply auraescript/src/bin/main.rs:32]: Put back, by I had to suppress a little as there is dead code that we will probably use and I couldn't get tonic to take the attributes on the generated code

File: auraescript/Cargo.toml (hunk 6 above)
Comment (future-highway) [reply auraescript/Cargo.toml:5]: of course I messed up copy pasting, lol

**Review by dmah42 — APPROVED** [review]

**Review by dmah42 — APPROVED** [review]

**Pull request merged**